Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -201,6 +201,316 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
+ * \param ion_lev       : Pointer to array of particle ionization level. This is
+                          required to have the charge of each macroparticle
+                          since q is a scalar. For non-ionizable species,
+                          ion_lev is a null pointer.
+ * \param rho_fab       : FArrayBox of charge density, either full array or tile.
+ * \param np_to_deposit : Number of particles for which charge is deposited.
+ * \param dx            : 3D cell size
+ * \param xyzmin        : Physical lower bounds of domain.
+ * \param lo            : Index lower bounds of domain.
+ * \param q             : species charge.
+ * \param n_rz_azimuthal_modes: Number of azimuthal modes when using RZ geometry.
+ * \param cost: Pointer to (load balancing) cost corresponding to box where present particles deposit current.
+ * \param load_balance_costs_update_algo: Selected method for updating load balance costs.
+ */
+template <int depos_order>
+void doChargeDepositionSharedShapeN (const GetParticlePosition& GetPosition,
+                                     const amrex::ParticleReal * const wp,
+                                     const int * const ion_lev,
+                                     amrex::FArrayBox& rho_fab,
+                                     const amrex::IntVect& ix_type,
+                                     const long np_to_deposit,
+                                     const std::array<amrex::Real,3>& dx,
+                                     const std::array<amrex::Real, 3> xyzmin,
+                                     const amrex::Dim3 lo,
+                                     const amrex::Real q,
+                                     const int n_rz_azimuthal_modes,
+                                     amrex::Real* cost,
+                                     const long load_balance_costs_update_algo,
+                                     const amrex::DenseBins<WarpXParticleContainer::ParticleType>& a_bins,
+                                     const amrex::Box& box,
+                                     const amrex::Geometry& geom,
+                                     const amrex::IntVect& a_tbox_max_size)
+{
+    using namespace amrex;
+
+#if !defined(AMREX_USE_GPU)
+    amrex::ignore_unused(ix_type, cost, load_balance_costs_update_algo, a_bins, box, geom, a_tbox_max_size);
+#endif
+
+    // Whether ion_lev is a null pointer (do_ionization=0) or a real pointer
+    // (do_ionization=1)
+    const bool do_ionization = ion_lev;
+    const amrex::Real dzi = 1.0_rt/dx[2];
+#if defined(WARPX_DIM_1D_Z)
+    const amrex::Real invvol = dzi;
+#endif
+#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+    const amrex::Real dxi = 1.0_rt/dx[0];
+    const amrex::Real invvol = dxi*dzi;
+#elif defined(WARPX_DIM_3D)
+    const amrex::Real dxi = 1.0_rt/dx[0];
+    const amrex::Real dyi = 1.0_rt/dx[1];
+    const amrex::Real invvol = dxi*dyi*dzi;
+#endif
+
+#if (AMREX_SPACEDIM >= 2)
```

Comment:
```suggestion
#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ) || defined(WARPX_DIM_3D)
```
Since we're moving toward `AMREX_SPACEDIM == 3` being true for all builds